sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/IntervalExpressionsSuite.scala

```diff
@@ -535,4 +533,195 @@ class IntervalExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360))),
+      ((Duration.ofDays(9999), BigDecimal(0.0001)),
+        Array(Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L),
+          Duration.ofSeconds(8639136000000L), Duration.ofSeconds(8639136000000L)),
+        Array(Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360),
+          Duration.ofMillis(86391360), Duration.ofMillis(86391360)))
+    ).foreach { case ((duration, num), divideExpected, multiplyExpected) =>
+      DataTypeTestUtils.dayTimeIntervalTypes.zip(divideExpected)
+        .foreach { case (dt, result) =>
+          checkEvaluation(DivideDTInterval(Literal.create(duration, dt), Literal(num)), result)
+        }
+      DataTypeTestUtils.dayTimeIntervalTypes.zip(multiplyExpected)
+        .foreach { case (dt, result) =>
+          checkEvaluation(MultiplyDTInterval(Literal.create(duration, dt), Literal(num)), result)
+        }
+    }
+  }
+
+  test("SPARK-35778: Check multiply/divide of year-month intervals of any fields by numeric 1") {
+    // we have guaranteed YearMonthIntervalType(YEAR, MONTH) multiply/divide numeric is right
+    Seq(
+      (Period.ofMonths(0), 10),
+      (Period.ofMonths(13), 1),
+      (Period.ofMonths(-200), 1),
+      (Period.ofYears(100), -1.toByte),
+      (Period.ofYears(1), 2.toShort),
+      (Period.ofYears(-1), -3),
+      (Period.ofMonths(-1000), 0.5f),
+      (Period.ofYears(1000), 100d),
+      (Period.ofMonths(2), BigDecimal(0.1))
+    ).foreach { case ((period, num)) =>
+      DataTypeTestUtils.yearMonthIntervalTypes.foreach { dt =>
+        val expectPeriod = dt.endField match {
+          case YEAR => Period.ofYears(period.normalized().getYears)
+          case MONTH => Period.ofYears(period.getYears).withMonths(period.getMonths)
+          case _ => period
+        }
+        val actualResult1 = MultiplyYMInterval(Literal.create(period, dt), Literal(num)).eval()
+        val expectResult1 = MultiplyYMInterval(Literal(expectPeriod), Literal(num)).eval()
+        assert(actualResult1 == expectResult1)
+        if (num != 0) {
+          val actualResult2 = DivideYMInterval(Literal.create(period, dt), Literal(num)).eval()
+          val expectResult2 = DivideYMInterval(Literal(expectPeriod), Literal(num)).eval()
+          assert(actualResult2 == expectResult2)
+        }
+      }
+    }
+  }
+
+  test("SPARK-35778: Check multiply/divide of year-month intervals of any fields by numeric 2") {
```

Comment:
This UT seems can cover all type field of YearMonthIntervalType

Reply:
[comment](https://github.com/apache/spark/pull/33080#issuecomment-868305049)